Source/WebKit/_WebKit_SwiftUI/Foundation+Extras.swift

```diff
@@ -35,16 +35,55 @@ func onNextMainRunLoop(do body: @escaping @MainActor () -> Void) {
 }
 
 struct AnyEquatable: Equatable {
-    let value: Any
-    private let equals: (Any) -> Bool
+    private enum Storage {
+        case value(Any, (Any) -> Bool)
+        case never
+    }
+
+    private let storage: Storage
+
+    var value: Any {
+        switch storage {
+        case let .value(value, _): value
+        case .never: ()
+        }
+    }
 
     init<E: Equatable>(_ value: E) {
-        self.value = value
-        self.equals = { ($0 as! E) == value }
+        self.storage = .value(value, { other in
+            guard let other = other as? E else {
+                assertionFailure()
+                return false
+            }
+
+            return other == value
+        })
+    }
+
+    /// Creates an `AnyEquatable` that is never equal to anything.
+    init() {
```

Comment:
I’m tempted to say you should call this `.nan` given that `AnyHashable() == AnyHashable()` is now `false`. Alternately… I think it’d be neat if we didn’t need this. I can see this being a future pitfall for someone.